pkg/vm/vm.go

```diff
@@ -22,37 +28,302 @@ type VirtualMachineConfig struct {
 type VirtualMachine struct {
 	Context       Context
 	MemoryManager *mem.MemoryManager
+	Step          uint64
 	Config        VirtualMachineConfig
 }
 
 // NewVirtualMachine creates a VM from the program bytecode using a specified config.
-func NewVirtualMachine(programBytecode *[]f.Element, config VirtualMachineConfig) (*VirtualMachine, error) {
+func NewVirtualMachine(programBytecode []*f.Element, config VirtualMachineConfig) (*VirtualMachine, error) {
 	manager, err := mem.CreateMemoryManager()
 	if err != nil {
 		return nil, fmt.Errorf("error creating new virtual machine: %w", err)
 	}
 
+	// 0 (programSegment) <- segment where the bytecode is stored
 	_, err = manager.Memory.AllocateSegment(programBytecode)
 	if err != nil {
 		return nil, fmt.Errorf("error loading bytecode: %w", err)
 	}
 
+	// 1 (executionSegment) <- segment where the stack trace will be stored
+	manager.Memory.Segments = append(manager.Memory.Segments, mem.EmptySegmentWithCapacity(10))
```

Comment:
Any thoughts how it'll be used?

Reply:
Yes, it doesn't make sense to have an "executionSegment" for stack trace only. So I joined both "dataSegment" with "executionSegment" and the stack trace is reflected in that same segment. I've updated the code to reflect this last sentences